WooCommerce/Classes/ViewRelated/Coupons/Add and Edit Coupons/AddEditCouponViewModel.swift

```diff
@@ -363,7 +442,9 @@ final class AddEditCouponViewModel: ObservableObject {
             return .couponCodeEmpty
         }
 
-        return nil
+        amountWarningTimer?.invalidate()
+        isDisplayingAmountWarning = false
+        return validatePercentageAmountInput(withWarning: false)
```

Comment:
Nit: should we only invalidate the timer and update the warning state if the `validatePercentageAmountInput` method returns a non-nil error?

Reply:
I wonder if doing so would raise the risk of flickering the warning colors into the form, since we would create a validation operation between losing the focus and canceling the warning state, does it makes sense or do you think it's an irrelevant risk?